mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -270,7 +270,7 @@ object MLUtils {
    */
   @Since("1.0.0")
   @Experimental
-  def kFold[T: ClassTag](rdd: RDD[T], numFolds: Int, seed: Int): Array[(RDD[T], RDD[T])] = {
+  def kFold[T: ClassTag](rdd: RDD[T], numFolds: Int, seed: Long): Array[(RDD[T], RDD[T])] = {
```

Comment:
It's likely source compatible, but I'm not sure if it's binary compatible.  It might be good to create a version for Int which calls into this version taking Long.